iceoryx_examples/ara_com/iox_ara_proxy.cpp

```diff
@@ -0,0 +1,135 @@
+        std::cout << "  Found no service(s), setting up asynchronous search with 'StartFindService'!" << std::endl;
+        auto callback = [&](com::ServiceHandleContainer<ara::com::ProxyHandleType> container,
+                            com::FindServiceHandle) -> void {
+            if (container.empty())
+            {
+                if (!maybeProxy->has_value())
+                {
+                    std::cout << "  No instance of service '" << MinimalProxy::m_serviceIdentifier
+                              << "' is available yet." << std::endl;
+                    return;
+                }
+                std::cout << "  Instance '" << maybeProxy->value().m_instanceIdentifier.c_str() << "' of service '"
+                          << MinimalProxy::m_serviceIdentifier << "' has disappeared." << std::endl;
+                maybeProxy->reset();
+                return;
+            }
+
+            for (auto& proxyHandle : container)
+            {
+                if (!maybeProxy->has_value())
+                {
+                    std::cout << "  Found instance of service: '" << MinimalProxy::m_serviceIdentifier << "', '"
+                              << proxyHandle.GetInstanceId().c_str() << "'" << std::endl;
+                    maybeProxy->emplace(proxyHandle);
+                }
+            }
+        };
+
+        auto handle = MinimalProxy::StartFindService(callback, searchString);
+        maybeHandle.emplace(handle);
+        std::cout << "  Waiting for instance called '" << searchString.c_str() << "' to become available.."
+                  << std::endl;
+    }
+
+    uint64_t addend1{0};
+    uint64_t addend2{0};
+
+    while (!iox::posix::hasTerminationRequested())
+    {
+        {
+            auto proxyGuard = maybeProxy.getScopeGuard();
+            if (proxyGuard->has_value())
+            {
+                auto& proxy = proxyGuard->value();
+                proxy.m_event.Subscribe(10U);
+
+                // Event
+                proxy.m_event.GetNewSamples(
+                    [](const auto& topic) { std::cout << "Receiving event: " << topic->counter << std::endl; });
+
+                // Field
+                proxy.m_field.GetNewSamples(
+                    [](const auto& field) { std::cout << "Receiving field: " << field->counter << std::endl; });
+
+                // Method
+                auto future = proxy.computeSum(addend1, addend2);
+                try
+                {
+                    auto result = future.get();
+                    std::cout << "Result of " << std::to_string(addend1) << " + " << std::to_string(addend2) << " is "
+                              << result.sum << std::endl;
+                }
+                catch (const std::future_error&)
+                {
+                    std::cout << "Empty future received, please start the 'iox-cpp-ara-skeleton'." << std::endl;
```

Comment:
This is not good as the contract should be that we get a value or nothing (and an error). This leads to the question how to propagate the error in case we do not get a value but also are not waiting (since the promise part does not exist anymore/yet).

Reply:
I didn't find a nicer solution that what is on the branch. Feel free to propose a better one.

Reply:
With `std::future` we have no choice. Alternative solutions would use a handcrafted future with additional API but is a lot more complex and out of scope for now. I decided I will donate futures and promises to iceoryx in the, well, future...